### ISSUE TITLE
Infinite loop in dart2js

### ISSUE THREAD

**Issue by fsc8000** [dart-lang/sdk#26493]
Repro:
```
python tools/test.py -mrelease -aia32 -cdart2js --dart2js-batch -rd8 --clear_browser_cache --write-debug-log --write-test-outcome-log --copy-coredumps --host-checked -t960 pkg/analyzer/tool/task_dependency_graph/check_test
```

It seems to have started with commit 144294fb2e8dee6a0b551473645c83286e95c39c which did not change anything in dart2js, but changed the input (analyzer)

**Comment by fsc8000** [comment]
@sigmundch @johnniwinther

**Comment by johnniwinther** [comment]
The infinite does not occur with a newer vm build. Closing assuming that this was caused by the reason problems in the vm.

**Comment by fsc8000** [comment]
Yes, it is possible that this is related to my fix baaea63e0a4b0e18dc423f967920b0bd42b5ea8a.

Did you re-enable the test again?

**Comment by johnniwinther** [comment]
Will do shortly.
